### PR TITLE
feat(TDI-39306) : Support of registered web app

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-39306

**What is the new behavior?**
The tMicrosoftCrmXXX can connect to a registered WebApp within user + client credentials.
Adal4j has been patched and is now part of libs_src.

The adal4j library has been added already updated. To see the patch, here is a PR on my own repository : https://github.com/ypiel-talend/azure-activedirectory-library-for-java/pull/1

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by ypiel-talend** [comment]
Adal4j ha sbeen added with its own .git inside that's why it was detected as a submodule.
It was listed within this command : git submodule--helper list (but no .submodule file in the root of the project).
I have removed it and added again deleted .git .gitignore files.
The library has been added already patched. To see it you can refer to a PR on my own repository:
https://github.com/ypiel-talend/azure-activedirectory-library-for-java/pull/1

File: main/plugins/org.talend.designer.components.libs/libs_src/adal4j/pom.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 	<modelVersion>4.0.0</modelVersion>
-	<groupId>com.microsoft.azure</groupId>
+	<groupId>org.talend.libraries</groupId>
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.libs/libs_src/adal4j/pom.xml]: why the `groupId` is changed?

Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/adal4j/pom.xml]: Since it is in libs_src now.

Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.libs/libs_src/adal4j/pom.xml]: It was a misunderstanding with Pierre.

**Pull request merged**